Sources/BuildServerProtocol/BuildTargets.swift

```diff
@@ -135,3 +135,75 @@ public struct BuildTargetCapabilities: Codable, Hashable {
     self.canRun = canRun
   }
 }
+
+/// The build target sources request is sent from the client to the server to
+/// query for the list of text documents and directories that are belong to a
+/// build target. The sources response must not include sources that are
+/// external to the workspace.
+public struct BuildTargetSources: RequestType, Hashable {
+  public static let method: String = "buildTarget/sources"
+  public typealias Response = BuildTargetSourcesResult
+
+  public var targets: [BuildTargetIdentifier]
+
+  public init(targets: [BuildTargetIdentifier]) {
+    self.targets = targets
+  }
+}
+
+public struct BuildTargetSourcesResult: ResponseType, Hashable {
+  public var items: [SourcesItem]
+}
+
+public struct SourcesItem: Codable, Hashable {
```

Comment:
Normally I like matching the protocol type names, but having this almost match `SourceItem` really threw me off.  This name isn't serialized anywhere, so how about changing it to something like `BuildTargetSources` so our API is clearer?  I've made similar exceptions for type names in the LSP support when they are unclear or ambiguous with things already in Swift.

Reply:
Fine by me. I'll change `OutputsItem` to match.

Reply:
We talked about this: `BuildTargetSources` conflicts with the request.  I agreed to keep the current names.